docs/extensions.md

```diff
@@ -0,0 +1,308 @@
+  - the  `tekton-dashboard-display-name: Hello` annotation make the extension appear in the side nav under the `Hello` name
+  - the `tekton-dashboard-endpoints: sample` annotation allows proxying requests to the extension by the Tekton Dashboard
+
+You can verify that the extension backend is working by hitting the path `/v1/extensions/sample-extension/sample`, it should display the `Hello Tekton Dashboard !` message.
+
+![Service based extension backend](./extensions-service-based-backend.png)
+
+The error message happens because the extension doesn't have frontend code yet, [next step](#example-add-and-serve-frontend-code) will guide you through adding frontend code to your extension.
+
+### Example: Add and serve frontend code
+
+The Tekton Dashboard is developed using [React](https://reactjs.org/), therefore the extension frontend code will need to use React too.
+
+All you need to do is provide a js module exporting your `Component`, this component will be loaded at runtime by the Tekton Dashboard host and injected into the page.
+
+For this to work, you will need to create your component using the shared objects provided by the host, namely you won't `import React, { Component } from 'react';` but use `window.React` instead.
+
+To deploy the service based extension with frontend code run the following command:
+
+```bash
+cat <<EOF | kubectl apply -n tekton-pipelines -f -
+kind: Service
+apiVersion: v1
+metadata:
+  name: sample-extension
+  labels:
+    app: sample-extension
+    tekton-dashboard-extension: "true"
+  annotations:
+    tekton-dashboard-display-name: Hello
+    tekton-dashboard-endpoints: sample.bundle
+    tekton-dashboard-bundle-location: bundle
+spec:
+  ports:
+  - port: 3000
+    targetPort: 3000
+  selector:
+    app: sample-extension
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: sample-extension
+  labels: 
+    app: sample-extension
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: sample-extension
+  template:
+    metadata:
+      labels:
+        app: sample-extension
+    spec:
+      containers:
+      - name: node
+        image: node:latest
+        ports:
+        - containerPort: 3000
+        command:
+        - bash
+        args:
+        - -c
+        - |
+          cat <<EOF > frontend.js
+          const React = window.React;
+          class Extension extends React.Component {
+            state = {
+              message: 'Loading ...'
+            };
+            componentDidMount() {
+              fetch('/v1/extensions/sample-extension/sample')
+                .then(response => response.text())
+                .then(message => this.setState({ message }));
+            };
+            render() {
+              const { message } = this.state;
+              return React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("h1", null, message));
```

Comment:
Can we simplify this? It's a little distracting, especially the `/*#__PURE__*/` magic comment since the React component isn't the focus of the example.

```suggestion
              return React.createElement("h1", null, message);
```